workflow-ui/container-android/src/main/java/com/squareup/workflow1/ui/modal/ModalContainer.kt

```diff
@@ -48,24 +52,46 @@ public abstract class ModalContainer<ModalRenderingT : Any> @JvmOverloads constr
     WorkflowLifecycleOwner.get(this)
   }
 
+  /**
+   * Connects the child `SavedStateRegistry` of each dialog layer to the `SavedStateRegistry`
+   * for this view.
+   */
+  private val stateRegistryAggregator = StateRegistryAggregator(
+    onWillSave = { aggregator ->
+      dialogs.forEach { ref ->
+        ref.stateRegistryOwner.let {
+          aggregator.saveRegistryController(it.key, it.controller)
+        }
+      }
+    },
+    onRestored = { aggregator ->
+      dialogs.forEach { ref ->
+        // We're only allowed to restore from an INITIALIZED state, but this callback can also be
```

Comment:
We have copy/pasted this fix in both places we use the aggregator. Can we build it in? I think we'd have to change it to depend on `KeyedStateRegistryOwner` directly. Maybe that's fine? Or vice versa, make the aggregator an implementation detail of `KSRO`.